modules/raft/src/main/java/org/apache/ignite/raft/jraft/core/IteratorImpl.java

```diff
@@ -135,6 +139,19 @@ protected void runTheRestClosureWithError() {
         }
     }
 
+    void runTheRestClosureWithShutdownException() {
+        Exception shutdownException = new ShutdownException();
```

Comment:
One exception per Raft group doesn't seem to be too much. I'm not sure whether we need a stack trace now, but, even for debugging purposes, it might be useful to have one when this doesn't incur a noticeable performance penalty (as in this case).

Reply:
Makes sense.